internal/compiler-bridge/src/main/scala/xsbt/ExtractUsedNames.scala

```diff
@@ -86,16 +86,14 @@ class ExtractUsedNames[GlobalType <: CallbackGlobal](val global: GlobalType) ext
      *     https://github.com/sbt/sbt/issues/1237
      *     https://github.com/sbt/sbt/issues/1544
      */
-    private val inspectedOriginalTrees = collection.mutable.Set.empty[Tree]
+    private val inspectedTrees = collection.mutable.Set.empty[Tree]
 
-    override def traverse(tree: Tree): Unit = tree match {
-      case MacroExpansionOf(original) if inspectedOriginalTrees.add(original) =>
-        handleClassicTreeNode(tree)
-        handleMacroExpansion(original)
-        super.traverse(tree)
-      case _ =>
-        handleClassicTreeNode(tree)
+    override def traverse(tree: Tree): Unit = {
+      if (!inspectedTrees.contains(tree)) {
+        if (handleTree(tree))
+          inspectedTrees += tree
         super.traverse(tree)
```

Comment:
sorry @eed3si9n it's not. Going to fix. Looks like there is no test to discover it.